frontend/src/metabase/questions/components/QuestionActivityTimeline.jsx

```diff
@@ -0,0 +1,87 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { t } from "ttag";
+import { connect } from "react-redux";
+import _ from "underscore";
+
+import { revertToRevision } from "metabase/query_builder/actions";
+import { getRevisionDescription } from "metabase/lib/revisions";
+
+import Revision from "metabase/entities/revisions";
+import Timeline from "metabase/components/Timeline";
+import ActionButton from "metabase/components/ActionButton";
+
+QuestionActivityTimeline.propTypes = {
+  question: PropTypes.object,
+  className: PropTypes.string,
+  revisions: PropTypes.array,
+  revertToRevision: PropTypes.func.isRequired,
+};
+
+function QuestionActivityTimeline({
+  question,
+  className,
+  revisions,
+  revertToRevision,
+}) {
+  const events = revisions.map((revision, index) => {
+    // const canRevert = question.canWrite();
```

Comment:
intentionally commenting this out for now, cuz the giant red REVERT button is kinda ugly. will deal with that later.